packages/fx-core/src/plugins/resource/appstudio/appStudio.ts

```diff
@@ -189,21 +188,29 @@ export namespace AppStudioClient {
         appDefinition.colorIcon = result.colorIconUrl;
         appDefinition.outlineIcon = result.outlineIconUrl;
       }
-      const response = await requester.post(
-        `/api/appdefinitions/${teamsAppId}/override`,
-        appDefinition
-      );
-      if (response && response.data) {
-        const app = <IAppDefinition>response.data;
-
-        if (app && app.teamsAppId && app.teamsAppId === teamsAppId) {
-          return app;
-        } else {
-          await logProvider?.error(
-            `teamsAppId mismatch. Input: ${teamsAppId}. Got: ${app.teamsAppId}`
+      let retry = 0;
```

Comment:
Nit: To reduce duplicated code, better to leverage https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/src/plugins/resource/bot/utils/retryHandler.ts